v3/src/components/graph/hooks/use-dot-plot.ts

```diff
@@ -0,0 +1,128 @@
+import { useCallback } from "react"
+import { ScaleBand, ScaleLinear } from "d3"
+import { useMemo } from "use-memo-one"
+import { useDataSetContext } from "../../../hooks/use-data-set-context"
+import { computePrimaryCoord, determineBinForCase, adjustCoordForStacks,
+         computeBinPlacements, computeSecondaryCoord} from "../utilities/dot-plot-utils"
+import { useGraphContentModelContext } from "./use-graph-content-model-context"
+import { useGraphDataConfigurationContext } from "./use-graph-data-configuration-context"
+import { useGraphLayoutContext } from "./use-graph-layout-context"
+import { AxisPlace } from "../../axis/axis-types"
+import { GraphAttrRole } from "../../data-display/data-display-types"
+import { setPointSelection } from "../../data-display/data-display-utils"
+import { useDataDisplayAnimation } from "../../data-display/hooks/use-data-display-animation"
+import { SubPlotCells } from "../models/sub-plot-cells"
+import { PixiPoints } from "../../data-display/pixi/pixi-points"
+
+export const useDotPlot = (pixiPoints?: PixiPoints) => {
+  const graphModel = useGraphContentModelContext()
+  const isHistogram = graphModel.pointDisplayType === "histogram"
+  const dataConfig = useGraphDataConfigurationContext()
+  const dataset = useDataSetContext()
+  const layout = useGraphLayoutContext()
+  const subPlotCells = useMemo(() => new SubPlotCells(layout, dataConfig), [dataConfig, layout])
+  const { secondaryNumericScale } = subPlotCells
+  const primaryAttrRole = dataConfig?.primaryRole ?? "x"
+  const primaryIsBottom = primaryAttrRole === "x"
+  const secondaryPlace = primaryIsBottom ? "left" : "top"
+  const secondaryAttrRole: GraphAttrRole = primaryAttrRole === "x" ? "y" : "x"
+  const extraPrimaryRole = primaryIsBottom ? "topSplit" : "rightSplit"
+  const extraPrimaryPlace = primaryIsBottom ? "top" : "rightCat"
+  const extraPrimaryAttrID = dataConfig?.attributeID(extraPrimaryRole) ?? ""
+  const extraPrimaryAxisScale = layout.getAxisScale(extraPrimaryPlace) as ScaleBand<string>
+  const numExtraPrimaryBands = Math.max(1, extraPrimaryAxisScale?.domain().length ?? 1)
+  const extraSecondaryRole = primaryIsBottom ? "rightSplit" : "topSplit"
+  const extraSecondaryAttrID = dataConfig?.attributeID(extraSecondaryRole) ?? ""
+  const primaryAttrID = dataConfig?.attributeID(primaryAttrRole) ?? ""
+  const primaryPlace: AxisPlace = primaryIsBottom ? "bottom" : "left"
+  const primaryAxisScale = layout.getAxisScale(primaryPlace) as ScaleLinear<number, number>
+  const pointDiameter = 2 * graphModel.getPointRadius()
+  const secondaryAttrID = dataConfig?.attributeID(secondaryAttrRole) ?? ""
+  const secondaryAxisScale = layout.getAxisScale(secondaryPlace) as ScaleBand<string>
```

Comment:
That's a very sensible suggestion but I tried it and it causes things to break for somewhat mysterious reasons. I'm going to skip this change, but will add a comment about how we should be using `getBandScale` here instead of `getAxisScale`, why we're not yet, etc.